.github/workflows/js-css-lint-test.yml

```
@@ -74,6 +78,20 @@ jobs:
         run: npm i -w assets -w storybook -w tests/js -w tests/e2e --include-workspace-root
       - name: JS Lint
         run: npm run lint:js
+  typecheck:
+    name: Type check
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+          cache: npm
+      - name: npm install
+        run: npm i -w assets -w storybook -w tests/js -w tests/e2e --include-workspace-root
```

We should run `npm ci` here to keep it deterministic.

```suggestion
        run: npm ci
```

---

I wasn't aware that we weren't using `npm ci` in all actions. It isn't just about removing node_modules if present but is more strict about what is installed which is what we want in CI, and I agree that we shouldn't assume that the two files are in sync.